### PR TITLE
fixed extra callback calls in while using cue to the soundFiles

### LINKED ISSUE #443: callback in addCue is called multiple times per cue time
p5 version: 1.0.0
p5.sound version: 0.3.12
Firefox 74.0.1, Ubuntu 19.10

The `callback` in `addCue(time, callback, [value])` is being called multiple times per cue time. The `value` remains the same until the next cue time.

Is this the desired behaviour?

I would have expected the `callback` to be called only once per cue time.

To see this, go to to the [online example](https://p5js.org/reference/#/p5.SoundFile/addCue) for `addCue` and modify it slightly as follows:
```
let mySound;
let numCalls;
function preload() {
  mySound = loadSound('assets/Damscray_DancingTiger.mp3');
}

function setup() {
  let cnv = createCanvas(100, 100);
  cnv.mousePressed(canvasPressed);
  background(220);
  text('tap to play', 10, 20);
  
  // schedule calls to changeText
  mySound.addCue(0, changeText, "hello" );
  mySound.addCue(0.5, changeText, "hello," );
  mySound.addCue(1, changeText, "hello, p5!");
  mySound.addCue(1.5, changeText, "hello, p5!!");
  mySound.addCue(2, changeText, "hello, p5!!!!!");

  mySound.onended(() => console.log(numCalls));
}

function changeText(val) {
  background(220);
  text(val, 10, 20);
  ++numCalls;
}

function canvasPressed() {
  numCalls = 0;
  mySound.play();
}
```

I get numCalls to be 44 or 45 and not 5. In Google Chrome it is 600 or so.

### PR DESCRIPTION
**Pull request by endurance21**
fixes #443

### CODE REVIEW

**Comment by endurance21** [comment]
@therewasaguy  please have a look on this pr !

**Comment by endurance21** [comment]
@montoyamoraga 
please have a look on this pr !

**Comment by endurance21** [comment]
Sounds great! I will be adding a unit test for the same :)

**Comment by endurance21** [comment]
I have added the tests for the same !
Selection_106
@therewasaguy  have a look at it ! 🙂

File: test/tests/p5.SoundFile.js [hunk 1]
```diff
@@ -115,6 +115,30 @@ define(['chai'], function(chai) {
...
+
+        sf.addCue(audioLength/5,callback); 
+        sf.addCue(audioLength*2/5,callback); 
+        sf.addCue(audioLength*3/5,callback); 
+        sf.addCue(audioLength*4/5,callback); 
+
+        sf.play();
+
+        sf.onended(()=>{
+          expect(numberOfCuesCalls).to.equal(4);
```
Comment (therewasaguy) [inline test/tests/p5.SoundFile.js:134]: awesome!

**Comment by endurance21** [comment]
looks great to me !

**Pull request merged**